pkg/util/kubeobjects/secret/replicate.go

```diff
@@ -0,0 +1,29 @@
+package secret
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Replicate will only create the secret once, doesn't mean for keeping the secret up to date
+func Replicate(ctx context.Context, query QueryObject, sourceSecretName, targetSecretName, sourceNs, targetNs string) error { //nolint:revive
```

Comment:
i once spent hours on a bug that turned out to be mixed up name and namespace args. my preventative measure at the time was  to use `client.ObjectKey`. i propose to do the same for this API. it's what the query needs anyways